var/spack/repos/builtin/packages/wgrib2/package.py

```diff
@@ -132,7 +132,13 @@ def inject_flags(self, name, flags):
 
         return (flags, None, None)
 
-    flag_handler = inject_flags
+    @property
```

Comment:
`flag_handler` is a method, so I don't think this will work to propagate compiler flags into `wgrib2`.

Reply:
`flag_handler` is defined as a `@property` in `PackageBase` ([ref](https://github.com/spack/spack/blob/develop/lib/spack/spack/package_base.py#L2176-L2184)), so this is needed to bind `self` to the method before returning it as a callable object. Otherwise MyPy errors about `flag_handler` being assigned a callable with a different signature, which I presume is a valid error.

It probably should just be a method, but making it one triggered more MyPy errors in other packages, so :shrug:

Reply:
New patch lifts `flag_handler` and a few related `@classmethod`'s into proper methods (instead of using Python trickery for nearly the same result).